uniflow/op/extract/split/recursive_character_splitter.py

```diff
@@ -0,0 +1,147 @@
+"""Recursive character split op."""
+
+import copy
+import re
+from typing import Iterable, List, Optional, Sequence
+
+from uniflow.node import Node
+from uniflow.op.op import Op
+
+
+class RecursiveCharacterSplitter(Op):
+    """Recursive character splitter class."""
+
+    default_separators = ["\n\n", "\n", " ", ""]
+
+    def __init__(
+        self,
+        name: str,
+        chunk_size: int = 1024,
+        chunk_overlap_size: int = 0,
+        separators: Optional[List[str]] = None,
+    ) -> None:
+        """Recursive Splitter Op Constructor
+
+        Args:
+            name (str): Name of the op.
+            chunk_size (int): Maximum size of chunks to return.
+            chunk_overlap_size (int): Overlap in characters between chunks.
+            separators (List[str]): Separators to use.
+            keep_separator: Whether to keep the separator in the chunks.
+        """
+        super().__init__(name)
+        self._chunk_size = chunk_size
+        self._chunk_overlap_size = chunk_overlap_size
+        self._separators = separators or self.default_separators
+
+    def __call__(self, nodes: Sequence[Node]) -> Sequence[Node]:
+        """Run Model Op.
+
+        Args:
+            nodes (Sequence[Node]): Nodes to run.
+            separators(List[str]): separators for split.
+
+        Returns:
+            Sequence[Node]: Nodes after running.
+        """
+        output_nodes = []
+        for node in nodes:
+            value_dict = copy.deepcopy(node.value_dict)
+            text = value_dict["text"]
+            text = self.recursive_splitter(text.strip(), self._separators)
+            output_nodes.append(
+                Node(
+                    name=self.unique_name(),
+                    value_dict={"text": text},
+                    prev_nodes=[node],
+                )
+            )
+        return output_nodes
+
+    def recursive_splitter(self, text: str, separators: List[str]) -> List[str]:
+        """Split incoming text and return chunks."""
```

Comment:
nit: can we add missing Args and Returns in the docstring. Meanwhile, could you please add a bit more comment regarding what is a recursive splitter and how it works to ease the review. Sorry. I am not on top of how it works.